### PR TITLE
Invite Settings

### PR DESCRIPTION
**Pull request by ThetaDR**
Signed-off-by: Vyacheslav Tumanov <me@slavatumanov.me>

# Contribution checklist

## Brief description
https://front.hc.engineering/workbench/platform/tracker/TSK-545
## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [ ] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [x] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: plugins/login-resources/src/index.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@
 //
 
 import LoginApp from './components/LoginApp.svelte'
-import InviteLink from './components/InviteLink.svelte'
+import InviteLink from '../../workbench-resources/src/components/InviteLink.svelte'
```
Comment (BykhovDenis) [inline plugins/login-resources/src/index.ts]: Please remove it here

File: plugins/login-resources/src/plugin.ts [hunk 2]
```diff
@@ -50,6 +50,8 @@ export default mergeIds(loginId, login, {
     ChangeAccount: '' as IntlString,
     WorkspaceNameRule: '' as IntlString,
     LinkValidHours: '' as IntlString,
+    EmailMask: '' as IntlString,
```
Comment (BykhovDenis) [inline plugins/login-resources/src/plugin.ts:59]: Please move it to workbench

File: plugins/login-assets/lang/en.json [hunk 3]
```diff
@@ -29,6 +29,8 @@
     "NotSeeingWorkspace": "Not seeing your workspace?",
     "WorkspaceNameRule": "The workspace name can contains lowercase letters, numbers, and symbols !@#%&^-",
     "LinkValidHours": "Link valid (hours):",
+    "EmailMask": "Email mask:",
```
Comment (BykhovDenis) [inline plugins/login-assets/lang/en.json:38]: Please move it to workbench

File: plugins/login-assets/lang/ru.json [hunk 4]
```diff
@@ -29,6 +29,8 @@
     "NotSeeingWorkspace": "Не видите ваше рабочее пространство?",
     "WorkspaceNameRule": "Название рабочего пространства должно состояить из строчных латинских букв, цифр и символов !@#%&^-",
     "LinkValidHours": "Ссылка действительна (часов):",
+    "EmailMask": "Маска для электронной почты:",
```
Comment (BykhovDenis) [inline plugins/login-assets/lang/ru.json:38]: Please move it to workbench

File: plugins/setting-resources/src/components/InviteSetting.svelte [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    if (existingInviteSettings.length === 0) {
+      await client.createDoc(
+        setting.class.InviteSettings,
+        preference.space.Preference,
+        newSettings
+      )
+    } else {
+      await client.updateDoc(
+        setting.class.InviteSettings,
+        preference.space.Preference,
```
Comment (BykhovDenis) [inline plugins/setting-resources/src/components/InviteSetting.svelte]: It's not preference, please create new space id in settings plugin

File: plugins/workbench-resources/src/components/InviteLink.svelte [hunk 6]
```diff
@@ -13,20 +13,58 @@
...
+  $: query.query(setting.class.InviteSettings, {}, (set) => {
+    if (set !== undefined && set.length > 0) {
+      expHours = set[0].expirationTime
+      emailMask = set[0].emailMask
+      limit = set[0].limit
+    }
+  })
+
+
+  async function setInviteSettings () {
```
Comment (BykhovDenis) [inline plugins/workbench-resources/src/components/InviteLink.svelte]: Extra

**Review by BykhovDenis — CHANGES_REQUESTED** [review]

File: packages/presentation/src/components/Members.svelte [hunk 7]
```diff
@@ -15,7 +15,7 @@
 <script lang="ts">
   import contact, { Employee, EmployeeAccount } from '@hcengineering/contact'
   import { Account, DocumentQuery, Ref, SortingOrder, Space } from '@hcengineering/core'
-  import login from '@hcengineering/login'
+  import workbench from '@hcengineering/workbench'
```
Comment (haiodo) [inline packages/presentation/src/components/Members.svelte]: @BykhovDenis @ThetaDR I suppose presentation should not know about any of login or even workspace, even for interfaces.

File: plugins/setting/src/index.ts [hunk 8]
```diff
@@ -76,6 +76,15 @@ export interface SettingsCategory extends Doc {
   secured: boolean
 }
 
+/**
+ * @public
+ */
+export interface InviteSettings extends Doc {
```
Comment (haiodo) [inline plugins/setting/src/index.ts]: Please extend Configuration base class, this kind of settings should be available only for owners of workspace.

File: plugins/workbench-resources/src/components/InviteLink.svelte [hunk 9]
```diff
@@ -13,20 +13,44 @@
...
 -->
 <script lang="ts">
-  import { Timestamp } from '@hcengineering/core'
-  import { copyTextToClipboard } from '@hcengineering/presentation'
-  import { Button, EditBox, getCurrentLocation, Label, Loading, locationToUrl, ticker } from '@hcengineering/ui'
-  import { getInviteLink } from '../utils'
-  import { createEventDispatcher, onMount } from 'svelte'
-  import login from '../plugin'
-  import InviteWorkspace from './icons/InviteWorkspace.svelte'
+  import { AccountRole, getCurrentAccount, Timestamp } from '@hcengineering/core'
```
Comment (haiodo) [inline plugins/workbench-resources/src/components/InviteLink.svelte]: I think all related to login/etc should be inside login module.

File: packages/presentation/src/components/Members.svelte (hunk 7 above)
Comment (BykhovDenis) [reply packages/presentation/src/components/Members.svelte]: Yes, it shouldn't. But it was implemented here, I think we should move it to another place in a separate PR

Comment (haiodo) [reply packages/presentation/src/components/Members.svelte]: ok

File: plugins/setting/src/index.ts (hunk 8 above)
Comment (haiodo) [reply plugins/setting/src/index.ts]: Ok, let's do in next PR.

**Review by BykhovDenis — APPROVED** [review]

**Pull request merged**